From e309ceda0ae3400c73c292f56aefcad5e0d5896f Mon Sep 17 00:00:00 2001
From: Nishanth Menon <nm@ti.com>
Date: Thu, 23 Jul 2015 17:43:31 -0500
Subject: [PATCH 19/25] cpufreq: voltdm: do a dummy opp setup as part of probe

As part of initial resource checks, we try and ensure that voltage
domain is ready with all required information. However to do that, we
also need to make sure that we have the OPP table initialized. Else, we
end up with:
  Adding alias for supply vdd,cpu0 -> vdd,4a0021c4.voltdm
  Adding alias for supply vbb,cpu0 -> vbb,4a0021c4.voltdm
  cpu cpu0: of_pm_voltdm_notifier_register: failed to get OPP, -19
  cpu cpu0: Failed to register cpu0 clock notifier: -19

To prevent this from happening, we do a dummy OPP initialization
and clean it up so that we can ensure all required voltage domain
information is also ready.

Signed-off-by: Nishanth Menon <nm@ti.com>
---
 drivers/cpufreq/cpufreq-voltdm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/cpufreq/cpufreq-voltdm.c b/drivers/cpufreq/cpufreq-voltdm.c
index 710899d..cf3a55a 100644
--- a/drivers/cpufreq/cpufreq-voltdm.c
+++ b/drivers/cpufreq/cpufreq-voltdm.c
@@ -281,8 +281,12 @@ static int voltdm_cpufreq_probe(struct platform_device *pdev)
 
 	np = cpu_dev->of_node;
 
+	/* OPPs might be populated at runtime, This is just a dummy setup */
+	dev_pm_opp_of_add_table(cpu_dev);
+
 	clk_nb = of_pm_voltdm_notifier_register(cpu_dev, np,
 						cpu_clk, "cpu0", &tmp);
+	dev_pm_opp_of_remove_table(cpu_dev);
 
 	if (IS_ERR(clk_nb)) {
 		ret = PTR_ERR(clk_nb);
-- 
2.7.0.rc3

